apis/refs/v1beta1/computerefs.go

```diff
@@ -147,47 +130,139 @@ func ResolveComputeSubnetwork(ctx context.Context, reader client.Reader, src cli
 	if region == "" {
 		return nil, fmt.Errorf("cannot get region from references ComputeSubnetwork %v: %w", key, err)
 	}
 
-	projectID, err := ResolveProjectID(ctx, reader, subnetObj)
-	if err != nil {
-		return nil, err
+	// Need to determine good targetField values, some common values: selfLink, name, id, email, or specific field name etc
+	switch targetField {
+	case "id":
+		return &ComputeSubnetworkRef{
+			External: fmt.Sprintf("projects/%s/regions/%s/networks/%s", projectID, region, resourceID),
+		}, nil
+	case "selfLink":
+		return &ComputeSubnetworkRef{
+			External: fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/networks/%s", projectID, region, resourceID),
+		}, nil
+	default:
+		return nil, fmt.Errorf("failed to resolve compute subnetwork reference: %s, target field: %s invalid", ref.Name, targetField)
 	}
-	return &ComputeSubnetworkRef{
-		External: fmt.Sprintf("projects/%s/regions/%s/subnetworks/%s", projectID, region, subnetID),
-	}, nil
 }
 
 type ComputeAddressRef struct {
-	/* The ComputeAddress selflink in the form "projects/{{project}}/regions/{{region}}/addresses/{{name}}" when not managed by KCC. */
+	/* An external value of a `ComputeAddress` resource, when not managed by KCC. */
 	External string `json:"external,omitempty"`
 	/* The `name` field of a `ComputeAddress` resource. */
 	Name string `json:"name,omitempty"`
 	/* The `namespace` field of a `ComputeAddress` resource. */
 	Namespace string `json:"namespace,omitempty"`
 }
 
+func ResolveComputeAddress(ctx context.Context, reader client.Reader, src client.Object, ref *ComputeAddressRef, targetField string) (*ComputeAddressRef, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.External != "" {
+		if ref.Name != "" {
+			return nil, fmt.Errorf("cannot specify both name and external on ComputeAddress reference")
+		}
+		return ref, nil
+	}
+
+	if ref.Name == "" {
+		return nil, fmt.Errorf("must specify either name or external on ComputeAddress reference")
+	}
+
+	key := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+	if key.Namespace == "" {
+		key.Namespace = src.GetNamespace()
+	}
+
+	computeAddress := &unstructured.Unstructured{}
+	computeAddress.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "compute.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "ComputeAddress",
+	})
+	if err := reader.Get(ctx, key, computeAddress); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("referenced ComputeAddress %v not found", key)
+		}
+		return nil, fmt.Errorf("error reading referenced ComputeAddress %v: %w", key, err)
+	}
+
+	resourceID, err := GetResourceID(computeAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	projectID, err := ResolveProjectID(ctx, reader, computeAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	region, _, _ := unstructured.NestedString(computeAddress.Object, "spec", "region")
```

Comment:
I may not have the full context, but I thought ComputeAddress CR doesn't have `region` field: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_computeaddresses.compute.cnrm.cloud.google.com.yaml?